src/shared/serverinfo.cpp

```diff
@@ -0,0 +1,507 @@
+            break;
+        case 2:
+            if(idx < 0)
+            {
+                intret(m_players.size());
+                return;
+            }
+            if( static_cast<size_t>( idx ) < m_players.size())
+            {
+                result(m_players[idx].data());
+            }
+            return;
+        case 3:
+            if(idx < 0)
+            {
+                intret(m_handles.size());
+                return;
+            }
+            if( static_cast<size_t>( idx ) < m_handles.size())
+            {
+                result(m_handles[idx].data());
+            }
+            return;
+    }
+}
+
+void serverinfo::writecfg( stream& file ) const
+{
+    file.printf("addserver %s %d %d %s %s %s %s\n",
+        m_name, m_port, m_priority,
+        escapestring(description()),
+        escapestring(m_authhandle),
+        escapestring(m_flags),
+        escapestring(m_branch)
+    );
+}
+
+void serverinfo::update( size_t len, void const* data, int serverdecay, int totalmillis, int lastreset )
+{
+    char text[MAXTRANS];
+    // const_cast: ucharbuf p is only read, but have no constness info
+    ucharbuf p( static_cast<uint8_t*>( const_cast<void*>( data ) ), len);
+    int millis = getint(p);
+    int rtt = clamp(totalmillis - millis, 0, min(serverdecay*1000, totalmillis));
+    if(millis >= lastreset && rtt < serverdecay*1000)
+    {
+        addping(rtt, millis);
+    }
+    m_lastinfo = totalmillis;
+    auto numplayers = getint(p);
+    int numattr = getint(p);
+    m_attr.clear();
+    loopj(numattr)
+    {
+        m_attr.emplace_back(getint(p));
+    }
+    int gver = m_attr.empty() ? 0 : m_attr[0];
+    getstring(text, p);
+    filterstring(m_map, text, false, true, true, false, sizeof( m_map ));
+    getstring(text, p);
+    filterstring(m_sdesc, text, true, true, true, false, MAXSDESCLEN+1);
+    m_players.clear();
+    m_handles.clear();
+    if(gver >= 227)
+    {
+        getstring(text, p);
+        filterstring(m_branch, text, true, true, true, false, MAXBRANCHLEN+1);
+    }
+    loopi(numplayers)
+    {
+        if(p.overread())
+        {
+            break;
+        }
+        getstring(text, p);
+        m_players.emplace_back(newstring(text));
+    }
+    if(gver >= 225)
+    {
+        loopi(numplayers)
+        {
+            if(p.overread())
+            {
+                break;
+            }
+            getstring(text, p);
+            m_handles.emplace_back(newstring(text));
+        }
+    }
+    s_sortedservers = false;
+}
+
+bool serverinfo::validate_resolve( char const* name, ENetAddress const& addr )
+{
+    if( m_name != name )
+    {
+        return false;
+    }
+    m_resolved = serverinfo::RESOLVED;
+    m_address.host = addr.host;
+    return true;
+}
+
+bool serverinfo::need_resolve( int& resolving )
+{
+    if( m_resolved == RESOLVED || m_address.host != ENET_HOST_ANY )
+    {
+        return false;
+    }
+    int ret = m_resolved == UNRESOLVED;
+    ++resolving;
```

Comment:
That shouldn't be done inside this method. The method does not make any use of the variable.